### PR TITLE
Past notifications should reflect current name of user

### LINKED ISSUE #2471: Make Names Reflect Current Name in Past Notifications
**What type(s) of user does this feature affect?**
 - volunteers
 - supervisors
 - admins

**Description**
Currently when someone updates their name, their name does not update in past notifications. We would like to change this so that the names appearing in notifications always reflect the current name of the person being referenced. 

**Screenshots of current behavior, if any**  
Here we see Whitney Block who used to be named Volunteer 1 still has a bunch of notifications referencing their old name:
Names Should Update When Changed

_QA Login Details:_  
[Link to QA site](https://casa-qa.herokuapp.com/)  

Login Emails: 
- volunteer1@example.com  view site as a volunteer
- supervisor1@example.com view site as a supervisor
- casa_admin1@example.com view site as an admin

password for all users: 123456

### PR DESCRIPTION
**Pull request by r-sierra**
Allows to get the users current name when showing the notification message

### What github issue is this PR for, if any?
Resolves #2471

### What changed, and why?
Notifications should reflect the current name of the user that created it. To accomplish this instead of passing the name of the user as a parameter we pass the user instance.

### How will this affect user permissions?
No permission affected

### How is this tested? (please write tests!) 💖💪
with tests

### CODE REVIEW

File: app/notifications/followup_notification.rb [hunk 1]
```diff
@@ -46,6 +46,14 @@ def message_without_note
...
 
   def message_heading
-    t(".message", created_by_name: params[:created_by_name])
+    t(".message", created_by_name: created_by_name)
+  end
+
+  def created_by_name
+    if params.key?(:created_by)
+      params[:created_by][:display_name]
+    else # keep backward compatibility with older notifications
```
Comment (compwron) [inline app/notifications/followup_notification.rb:55]: wonderful :)

File: app/notifications/followup_resolved_notification.rb [hunk 2]
```diff
@@ -22,10 +22,20 @@ def title
...
+    t(".message", created_by_name: created_by_name)
   end
 
   def url
     edit_case_contact_path(params[:followup][:case_contact_id], notification_id: record.id)
   end
+
+  private
+
+  def created_by_name
```
Comment (compwron) [inline app/notifications/followup_resolved_notification.rb:34]: duplicate code block- can we combine them?

File: spec/system/notifications/index_spec.rb [hunk 3]
```diff
@@ -12,19 +12,40 @@
...
+    it "lists my notifications" do
       sign_in admin
       visit notifications_path
 
       expect(page).to have_text(notification_message)
       expect(page).to have_text("Followup resolved")
       expect(page).not_to have_text(I18n.t(".notifications.index.no_notifications"))
     end
+
+    context "when volunteer changes its name" do
```
Comment (compwron) [inline spec/system/notifications/index_spec.rb:31]: Please add a test for old notifications being displayable

**Review by compwron — APPROVED** [review]

File: app/notifications/followup_resolved_notification.rb (hunk 2 above)
Comment (r-sierra) [reply app/notifications/followup_resolved_notification.rb:34]: Sure

File: spec/system/notifications/index_spec.rb (hunk 3 above)
Comment (r-sierra) [reply spec/system/notifications/index_spec.rb:31]: Not sure we can do that, as notifications fail if required params are not filled on save. I'll see what i can do

**Comment by compwron** [comment]
nice! :)

**Pull request merged**